test/features/opening_hours/domain/usecases/check_open_status_test.dart

```diff
@@ -7,25 +7,25 @@ import 'package:mockito/mockito.dart';
 
 import 'check_open_status_test.mocks.dart';
 
-@GenerateMocks([OpeningHoursRepository])
+@GenerateMocks([OpeningHoursRemoteDataSource])
 void main() {
-  late MockOpeningHoursRepository repository;
+  late MockOpeningHoursRemoteDataSource dataSource;
   late CheckOpenStatus getIsOpen;
 
   setUp(() {
-    repository = MockOpeningHoursRepository();
-    getIsOpen = CheckOpenStatus(repository: repository);
+    dataSource = MockOpeningHoursRemoteDataSource();
+    getIsOpen = CheckOpenStatus(dataSource: dataSource);
   });
 
-  test('should call repository', () async {
+  test('should call data source', () async {
     // arrange
-    when(repository.getIsOpen()).thenAnswer((_) async => const Right(true));
+    when(dataSource.isOpen()).thenAnswer((_) async => const Right(true));
 
     // act
     await getIsOpen(NoParams());
 
     // assert
-    verify(repository.getIsOpen());
-    verifyNoMoreInteractions(repository);
+    verify(dataSource.isOpen());
```

Comment:
add `.called(1)` to check for only a single call

```suggestion
    verify(dataSource.isOpen()).called(1);
```

Reply:
verify(dataSource.isOpen()) and verify(dataSource.isOpen()).called(1) is identical i believe. The .called(1) is implicit

Reply:
(Counter-intuitively) it doesn't work like that. Try verifying by duplicating the call to `getIsOpen` above; the test still passes. Only when `.called(1)` is added to the verify function is the test invalidated by two calls to `getIsOpen`

Reply:
For a later conversation